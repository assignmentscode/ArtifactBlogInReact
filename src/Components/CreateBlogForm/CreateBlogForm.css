.form-width {
  width: 50%;
  margin: auto;
}
.flex-outer li {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  align-content: space-around;
  margin-top: 5%;
  margin-bottom: 5%;
}
.flex-outer > li > label,
.flex-outer li p {
  flex: 1 0 120px;
  max-width: 220px;
}
 
.flex-outer > li > label + *{
  flex: 1 0 220px;
}
  .btn-submit {
    margin-left: 50%;
    margin-bottom: 1%;
    padding: 8px 16px;
    border: none;
    background: green;
    color: #f2f2f2;
    text-transform: uppercase;
    letter-spacing: .09em;
    border-radius: 8px;
  }

