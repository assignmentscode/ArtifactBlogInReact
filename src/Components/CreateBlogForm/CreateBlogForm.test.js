import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import CreateBlogForm from './CreateBlogForm';

describe('CreateBlogForm', ()=> {
    it('should render form for creating Blog Post', ()=> {
        const tree = renderer.create(<CreateBlogForm />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    const wrapper = shallow(<CreateBlogForm/>);
    it('should have 5 list items element', () => {
        expect(wrapper.find('li').getElements().length).toEqual(5);
    });
});