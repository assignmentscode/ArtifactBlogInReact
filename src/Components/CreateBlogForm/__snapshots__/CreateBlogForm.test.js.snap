// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateBlogForm should render form for creating Blog Post 1`] = `
<form>
  <ul
    class="flex-outer"
  >
    <li>
      <label
        for="date"
      >
        Date
      </label>
      <input
        id="date"
        type="date"
      />
    </li>
    <li>
      <label
        for="read-time"
      >
        Read Time
      </label>
      <input
        id="read-time"
        placeholder="Enter the read time for blog"
        type="text"
      />
    </li>
    <li>
      <label
        for="title"
      >
        Title
      </label>
      <input
        id="title"
        placeholder="Enter the Title of Post"
        type="text"
      />
    </li>
    <li>
      <label
        for="message"
      >
        Message
      </label>
      <textarea
        id="message"
        placeholder="Enter your message here"
        rows="6"
      />
    </li>
    <li>
      <button
        type="submit"
      >
        Submit
      </button>
    </li>
  </ul>
</form>
`;
